diff --git a/Makefile b/Makefile
index 5bafc73..e926877 100644
--- a/Makefile
+++ b/Makefile
@@ -37,7 +37,7 @@
 # so check 'nodejs' first.
 GIT ?= $(shell command -v git 2>/dev/null)
 GIT := $(GIT)
-NODEJS ?= $(shell { command -v nodejs || command -v node; } 2>/dev/null)
+NODEJS ?= $(shell { command -v node || command -v nodejs; } 2>/dev/null)
 NODEJS := $(NODEJS)
 WGET ?= $(shell command -v wget 2>/dev/null)
 WGET := $(WGET)
@@ -49,7 +49,7 @@ PYTHON3 ?= $(shell { command -v python3 || command -v python; } 2>/dev/null)
 PYTHON3 := $(PYTHON3)
 PYTHON2 ?= $(shell { command -v python2 || command -v python; } 2>/dev/null)
 PYTHON2 := $(PYTHON2)
-PYTHON ?= $(PYTHON2)
+PYTHON ?= $(PYTHON3)
 PYTHON := $(PYTHON)
 DOCKER ?= docker
 DOCKER := $(DOCKER)
diff --git a/tools/configure.py b/tools/configure.py
index 88fb6981..7524e664 100644
--- a/tools/configure.py
+++ b/tools/configure.py
@@ -14,18 +14,18 @@ import tempfile

 def detect_nodejs():
     try:
-        cmd = [ 'nodejs', '-e', 'console.log("test")' ]
+        cmd = [ 'node', '-e', 'console.log("test")' ]
         res = subprocess.check_output(cmd)
         if res[:4] == 'test'.encode('utf-8'):
-            return 'nodejs'
+            return 'node'
     except:
         pass

     try:
-        cmd = [ 'node', '-e', 'console.log("test")' ]
+        cmd = [ 'nodejs', '-e', 'console.log("test")' ]
         res = subprocess.check_output(cmd)
         if res[:4] == 'test'.encode('utf-8'):
-            return 'node'
+            return 'nodejs'
     except:
         pass

